06_predict_survival/nbconverted/plot_survival_pc_comparison.py

```diff
@@ -0,0 +1,459 @@
+
+# set list of PCA component numbers to look for
+pcs_list = [10, 100, 500, 1000, 5000]
+
+# order to plot data types in
+training_data_map = {
+    'expression': 'gene expression',
+    'me_27k': '27k methylation',
+    'me_450k': '450k methylation',
+}
+
+
+# In[3]:
+
+
+me_data_count_df = []
+
+for n_dim in pcs_list:
+    # load results into a single dataframe
+    me_pcs_dir = Path(me_results_dir, 'results_{}_pca'.format(n_dim))
+    me_results_df = su.load_survival_results(me_pcs_dir)
+    me_results_df.rename(columns={'identifier': 'cancer_type',
+                                  'fold_no': 'fold'}, inplace=True)
+    
+    me_count_df = (me_results_df[me_results_df.data_type == 'test']
+        .groupby(['cancer_type', 'training_data'])
+        .count()
+    )
+    problem_df = me_count_df[me_count_df['cindex'] != 16].copy()
+    drop_cancer_types = problem_df.index.get_level_values(0).unique().values
+    
+    me_all_results_df = au.compare_all_data_types(
+        me_results_df[~me_results_df.cancer_type.isin(drop_cancer_types)],
+        SIG_ALPHA,
+        identifier='cancer_type',
+        metric='cindex')
+    
+    print(n_dim)
+    num_reject = me_all_results_df.groupby(['training_data']).sum().reject_null
+    num_total = me_all_results_df.groupby(['training_data']).count().reject_null
+    
+    dim_count_df = (
+            pd.DataFrame(num_reject).rename(columns={'reject_null': 'num_reject'})
+        .merge(
+            pd.DataFrame(num_total).rename(columns={'reject_null': 'num_total'})
+        , left_index=True, right_index=True)
+    )
+    dim_count_df['n_dim'] = n_dim
+    dim_count_df['ratio'] = dim_count_df.num_reject / dim_count_df.num_total
+    me_data_count_df.append(dim_count_df)
+    
+me_data_count_df = pd.concat(me_data_count_df).reset_index()
+me_data_count_df.training_data.replace(to_replace=training_data_map, inplace=True)
+me_data_count_df.head(10)
+
+
+# In[4]:
+
+
+sns.set({'figure.figsize': (22, 5)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(1, 3)
+
+sns.pointplot(data=me_data_count_df, x='n_dim', y='num_reject',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[0])
+axarr[0].set_xlabel('Number of PCs')
+axarr[0].set_ylabel('Number of well-predicted cancer types')
+axarr[0].set_title('Well-predicted cancer type count vs. PC count')
+
+sns.pointplot(data=me_data_count_df, x='n_dim', y='num_total',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[1])
+axarr[1].set_xlabel('Number of PCs')
+axarr[1].set_ylabel('Total valid cancer types')
+axarr[1].set_title('Valid cancer type count vs. PC count')
+
+sns.pointplot(data=me_data_count_df, x='n_dim', y='ratio',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[2])
+axarr[2].set_xlabel('Number of PCs')
+axarr[2].set_ylabel('Ratio of well-predicted to total cancer type')
+axarr[2].set_title('Well-predicted/total ratio vs. PC count')
+
```

Comment:
Right, Casey and I don't think this plot is a very good reflection of performance at different PC counts, which is why we decided not to put it in the paper. The main reason is that it averages over cancer types - some of the cancer types have very small sample sizes so performance might jump around a lot or saturate really quickly, and when we just look at the average this results in weird trends.

When we break things out into separate cancer types, what we see makes a little more sense (at least to me), for instance these plots:

![image](https://user-images.githubusercontent.com/2345877/132545754-02b39a16-6617-40af-b02d-34fb2dcc6d97.png)

![image](https://user-images.githubusercontent.com/2345877/132545699-2201fca8-ccbc-467f-98a6-33a30147ecb8.png)

Here, it looks like either performance tends to increase a bit with added PCs (e.g. in LGG, and a bit in the pan-cancer plot) or it tends to saturate quickly and decrease (e.g. sarcoma or melanoma, maybe because the top PCs have most of the relevant info and later ones are essentially random/result in feature selection challenges). In the paper we're only going to show the pan-cancer results, and we'll put some of the individual cancer types in the supplement.